### DISCUSSION TITLE
Austenite grain reconstruction dependence on initial grain segmentation

### DISCUSSION THREAD

**Discussion by viviannh81** [mtex-toolbox/mtex#849] [Ask Anything]
Hello everyone, 

I recently tried to reconstruct prior austenite based on TEM ASTAR acquisition where I encountered problem where the reconstruction depends strongly on the initial grain segmentation.
[grains,ebsd.grainId] = calcGrains(ebsd('Ferrite'), 'angle', 3*degree);

I also identified this dependence for EBSD mapping reconstruction. As can be seen in the following image of prior austenite reconstruction, the EBSD mapping are for tempered martensite. The 2 reconstructions are based on 2 different criteria : 1° and 3° for initial grain segmentation, the reconstruction results are fairly different.

![image](https://user-images.githubusercontent.com/51818024/110668430-97cbeb80-81cb-11eb-8c2f-daca4d3feba5.png)

Is there a specific reason of the dependence? How do I pick the appropriate criteria for the initial grain segmentation?

Thanks!

**Comment by nyyssont** [comment]
Short answer: I would use a 3° angular threshold.

Long answer:
As you know, the crystallographic orientation of a martensitic lath is the result of a relatively rigidly defined rotation of the prior austenite orientation. This is called the orientation relationship. It follows that all of the boundary misorientations between laths originating from the same prior austenite grain can be defined from the orientation relationship alone.

Using the Kurdjumov-Sachs orientation relationship, Morito et al. determined the possible misorientations within a single packet to be as follows:

![Morito_misos](https://user-images.githubusercontent.com/10060198/110749638-ace36180-8241-11eb-8e31-965c092b3c5a.png)

You will note that within a packet, the V1-V4, V2-V5 and V3-V6 misorientations have a rather low disorientation angle, about 10°. This, in theory, is the lowest possible misorientation angle between laths.

In low-carbon lath martensite, however, the orientation relationship is usually in fact closer to the Greninger-Troiano relationship rather than K-S. In this case, the V1-V4 type disorientation angle becomes about 6 degrees. Considering grain segmentation in MTEX, a 3° threshold should still be enough to separate variants completely. However, this appears to not be the case, as shown by the effect of lowering the threshold angle in the image above. There are two factors complicating things:

1. The orientation relationship is, at best, a representative average value used to describe a large population of laths. In reality, the orientation relationship has considerable spread and some variants may in fact be best described by a wholly different representative average. This has the consequence that in some cases the V1-V4 disorientation angles become small enough that they cannot be distinguished based on angular threshold of 3°.
2. Annealing twins in the prior austenite microstructure share several martensitic variants between them. In many cases, the variants that form on different sides of the annealing twin boundary are crystallographically nearly equivalent and cannot be distinguished based on an angular threshold.

To summarize: in some cases, a 3° angular threshold is not low enough for complete segmentation of martensite. However, lowering it to 1° will cause additional problems: the accuracy of the measurement system becomes an issue, as well as local misorientations in the high-dislocation density martensite. Thus, the segmentation using 1° may be too powerful and result in a large amount of low-angle boundary misorientations that the reconstruction algorithms in MTEX cannot properly interpret.

Concerning the image above: both results may be equally valid. You might want to take a look at how each grain fits with the suggested prior austenite solution:

```
figure
plot(job.grains,job.grains.fit./degree)
```

It may be that there is no difference in fit for that area using either 1° or 3° angular threshold. This implies that the problematic grain boundaries are annealing twins, and in the vicinity of the twin boundary are variants that fit almost equally well to both solutions. In that case, it is impossible to nail down the annealing twin boundary exactly, using the tools that are available right now.

May I ask which reconstruction approach you are using? If you use Markov Clustering, you might want to try the "majority vote" approach instead to see if you get a different result:
```
job = parentGrainReconstructor(ebsd,grains) % define job
job.calcParent2Child; % get representative OR

job.calcGBVotes % compute votes from child grain boundaries
job.calcParentFromVote('minVotes',4)

for l = 1:2
job.calcGBVotes('noC2C') % compute votes ONLY from parent-child boundaries;
job.calcParentFromVote('minFit',5*degree) %compute parent orientations from votes
end
```
